### PR TITLE
📖 add settings for developing with Tilt and MachinePools

### PR DESCRIPTION
**Pull request by mboersma**
**What this PR does / why we need it**:

Documents the changes to `tilt-settings.json` that enable experimental MachinePool support. This borrows from a [CAPI doc section](https://cluster-api.sigs.k8s.io/developer/tilt.html#create-a-tilt-settingsjson-file), but I think it's better in this case to duplicate it locally so CAPZ docs are a bit more self-contained.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
📖 add settings for developing with Tilt and MachinePools
```

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -397,8 +417,8 @@ export SKIP_UPSTREAM_E2E_TESTS="false"
 [go.mod]: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/go.mod
 [kind]: https://sigs.k8s.io/kind
 [azure_cli]: https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
-[manifests]: /docs/manifests.md
```
Comment (mboersma) [inline docs/development.md:400]: This was a dead link that wasn't referenced.

**Comment by CecileRobertMichon** [comment]
@mboersma, @chewong is doing something similar in his MachinePools E2E PR, see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/659/files#diff-b67911656ef5d18c4ae36cb6741b7965R86

**Comment by mboersma** [comment]
> something similar in his MachinePools E2E PR

I guess patching it from the Makefile would obviate the need to document this. I'll comment over on #659 instead.

**Comment by mboersma** [comment]
> maybe you can build on top of his PR once it merges

Sure, I'll revisit this once #659 goes in.

**Pull request closed without merging**